### PR TITLE
Don't autofocus input on initial visit, only show tour when input focused

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/14783.

### CODE REVIEW

File: client/web/src/search/input/LazyMonacoQueryInput.test.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ describe('PlainQueryInput', () => {
                         copyQueryButton={false}
                         versionContext={undefined}
                         globbing={false}
+                        showOnboardingTour={false}
```
Comment (limitedmage) [inline client/web/src/search/input/LazyMonacoQueryInput.test.tsx:32]: Can we add a test where this is true and unit tests to check the autofocus?

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/input/SearchPageInput.tsx [hunk 2]
```diff
@@ -288,7 +287,7 @@ export const SearchPageInput: React.FunctionComponent<Props> = (props: Props) =>
                                 queryState={userQueryState}
                                 onChange={setUserQueryState}
                                 onSubmit={onSubmit}
-                                autoFocus={props.autoFocus !== false}
+                                autoFocus={showOnboardingTour ? !!tour.isActive() : props.autoFocus !== false}
```
Comment (lguychard) [inline client/web/src/search/input/SearchPageInput.tsx:290]:
```suggestion
                                autoFocus={showOnboardingTour ? tour.isActive() : props.autoFocus !== false}
```

Doesn't `tour.isActive()` already return a boolean?

File: client/web/src/integration/search-onboarding.test.ts [hunk 3]
```diff
@@ -84,6 +84,8 @@ describe('Search onboarding', () => {
     describe('Onboarding', () => {
         it('displays all steps in the language onboarding flow', async () => {
             await driver.page.goto(driver.sourcegraphBaseUrl + '/search')
+            await driver.page.waitForSelector('.monaco-editor .view-lines')
+            await driver.page.click('.monaco-editor .view-lines')
             await driver.page.waitForSelector('.tour-card')
             await driver.page.waitForSelector('.test-tour-language-button')
             await driver.page.click('.test-tour-language-button')
```
Comment (lguychard) [inline client/web/src/integration/search-onboarding.test.ts:91]: There appears to be a lot of copy-pasting in the steps of these tests. Could they be DRYed?

**Review by lguychard — APPROVED** [review]

File: client/web/src/search/input/LazyMonacoQueryInput.test.tsx (hunk 1 above)
Comment (attfarhan) [reply client/web/src/search/input/LazyMonacoQueryInput.test.tsx:32]: added integration test

**Pull request merged**